File: StudyGamePrograming08/SomeSpriteComponent.cpp
#include "SomeSpriteComponent.h"
#include "Actor.h"
#include "Game.h"
#include "Renderer.h"

SomeSpriteComponent::SomeSpriteComponent(Actor* owner, int drawOrder)
	: SpriteComponent(owner, drawOrder)
{}

void SomeSpriteComponent::SetSomeTextures(std::vector<std::string> texturefiles)
{
	for (auto file : texturefiles)
	{
		SetTexture(mOwner->GetGame()->GetRenderer()->GetTexture(file));
	}	
}

void SomeSpriteComponent::SelectTexture(std::string filename)
{
	SetTexture(mOwner->GetGame()->GetRenderer()->GetTexture(filename));
}

File: StudyGamePrograming08_VB/SomeSpriteComponent.vb
Public Class SomeSpriteComponent
	Inherits SpriteComponent

	' テクスチャ用の画像ファイルリスト
	Public TextureFiles As List(Of String)

	Sub New(ByRef owner As Actor, ByVal drawOrder As Integer)
		MyBase.New(owner, drawOrder)

	End Sub

	Public Sub SetSomeTextures(ByVal texturefiles As List(Of String))
		For Each file In texturefiles
			SetTexture(mOwner.GetGame().GetRenderer().GetTexture(file))
		Next
	End Sub
	Public Sub SelectTexture(ByVal filename As String)
		SetTexture(mOwner.GetGame().GetRenderer.GetTexture(filename))
	End Sub
End Class